core/domain/skill_domain.py

```diff
@@ -1776,3 +1776,151 @@ def from_dict(cls, skill_mastery_dict):
+
+        Args:
+            topic_name: str. The name of the topic.
+            skill_id: str. The id of the skill.
+            skill_description: str. The description of the skill.
+        """
+        self.require_topic_name_to_be_added(topic_name)
+        self.categorized_skills[topic_name]['uncategorized'].append(
+            ShortSkillSummary(skill_id, skill_description))
+
+    def add_subtopic_skill(
+        self, topic_name, subtopic_title, skill_id, skill_description):
+        """Adds a subtopic skill id and description for the given topic.
+
+        Args:
+            topic_name: str. The name of the topic.
+            subtopic_title: str. The title of the subtopic.
+            skill_id: str. The id of the skill.
+            skill_description: str. The description of the skill.
+        """
+        self.require_topic_name_to_be_added(topic_name)
+        self.require_subtopic_title_to_be_added(topic_name, subtopic_title)
+        self.categorized_skills[topic_name][subtopic_title].append(
+            ShortSkillSummary(skill_id, skill_description))
+
+    def require_topic_name_to_be_added(self, topic_name):
+        """Checks whether the given topic name is valid i.e. added to the
+        categorized skills dict.
+
+        Args:
+            topic_name: str. The name of the topic.
+
+        Raises:
+            ValidationError. Topic name is not added.
+        """
+        if not topic_name in self.categorized_skills:
+            raise utils.ValidationError(
+                'Topic name \'%s\' is not added.' % topic_name)
+
+    def require_subtopic_title_to_be_added(self, topic_name, subtopic_title):
+        """Checks whether the given subtopic name is added to the
+        categorized skills dict under the given topic name.
+
+        Args:
+            topic_name: str. The name of the topic.
+            subtopic_title: str. The name of the subtopic.
+
+        Raises:
+            ValidationError. Subtopic name is not added.
+        """
+        if not subtopic_title in self.categorized_skills[topic_name]:
+            raise utils.ValidationError(
+                'Subtopic title \'%s\' is not added.' % subtopic_title)
+
+    def to_dict(self):
+        """Returns a dictionary representation of this domain object."""
+        categorized_skills_dict = copy.deepcopy(self.categorized_skills)
+
+        for topic_name in categorized_skills_dict:
+            # The key 'uncategorized' will also be covered by this loop.
+            for subtopic_title in categorized_skills_dict[topic_name]:
+                categorized_skills_dict[topic_name][subtopic_title] = [
+                    short_skill_summary.to_dict() for short_skill_summary in
+                    categorized_skills_dict[topic_name][subtopic_title]
+                ]
+        return categorized_skills_dict
+
+
+class ShortSkillSummary:
+    """Domain object for a short skill summary. It is similar to SkillSummary
+    but has fewer fields.
```

Comment:
This is quite a vague description (the second sentence). Try to explain constructively what it is and how exactly it differs from SkillSummary.

Reply:
Done.